rosidl_generator_cpp/include/rosidl_generator_cpp/bounded_vector.hpp

```diff
@@ -454,6 +460,16 @@ class BoundedVector
     _Base::push_back(__x);
   }
 
+  template<typename ... Args>
+  void
+  emplace_back(Args && ... args)
+  {
+    if (size() >= _UpperBound) {
+      throw std::length_error("Exceeded upper bound");
+    }
+    _Base::emplace_back(std::forward<Args>(args)...);
```

Comment:
Instead of calling `_Base::emplace_back` here, you should be able to remove the if statement and instead just `return emplace(this->cend(), std::forward<Args>(args) ...)`, which helps de-duplicate the code here. 

I think this should also return a reference the emplaced object (which is standard behavior in C++17).

Reply:
By default, we are using c++14.
Maybe, change the return type to `auto`, don't remove the if and do `return _Base::emplace_back(std::forward<Args>(args)...);`.
It will return the correct type when being compiled with both cpp14 or cpp17.

Reply:
@ivanpauno @allenh1 Thanks for the recommendations!

I went with @ivanpauno 's suggestion, since I think maintaining API compatibility is probably more important.